spec/factories/requests.rb

```diff
@@ -10,5 +10,20 @@
     decision     { :undecided }
 
     workflow
+
+    trait :with_context do
+      after(:create) do |obj|
+        obj.update_attributes(:context =>
+          {
+            "headers"      => {"x-rh-identity"=> "eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6IjAzNjkyMzMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJqZG9lIiwiZW1haWwiOiJqZG9lQGFjbWUuY29tIiwiZmlyc3RfbmFtZSI6IkpvaG4iLCJsYXN0X25hbWUiOiJEb2UiLCJpc19hY3RpdmUiOnRydWUsImlzX29yZ19hZG1pbiI6ZmFsc2UsImlzX2ludGVybmFsIjpmYWxzZSwibG9jYWxlIjoiZW5fVVMifSwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMzM0MDg1MSIsImF1dGhfdHlwZSI6ImJhc2ljLWF1dGgiLCJhdXRoX3RpbWUiOjYzMDB9fX0="},
```

Comment:
You can use `encoded_user_hash`, which is defined in `spec/support/request_spec_helper.rb` as `x-rh-identity`:
```
"headers"      => {"x-rh-identity"=>encoded_user_hash},
```